File: Server/Matricula/src/java/com/sistema/LogicaNegocio/Grupo.java
package com.sistema.LogicaNegocio;
import java.util.ArrayList;
import java.util.List;
/**
 *
 * @author DavidTK1198
 */
public class Grupo {
    private int numero;
    private Ciclo ciclo;
    private String horario;
    private List<Alumno> estudiantes;
    private Profesor profesor;
    private Curso curso;
    private int idEntidad;

    public int getIdEntidad() {
        return idEntidad;
    }

    public void setIdEntidad(int idEntidad) {
        this.idEntidad = idEntidad;
    }
    private int cupo;
    private int disponible;

    public Grupo(int cupo,int disponible,int numero,String horario,Ciclo ciclo, Profesor profesor, Curso curso) {
        this.numero = numero;////
        this.horario = horario;
        this.profesor = profesor;
        this.curso = curso;
        estudiantes = new ArrayList();
        this.ciclo=ciclo;
        this.cupo=cupo;
        this.disponible=disponible;
    }

    public Ciclo getCiclo() {
        return ciclo;
    }

    public void setCiclo(Ciclo ciclo) {
        this.ciclo = ciclo;
    }

    public int getDisponible() {
        return disponible;
    }

    public void setDisponible(int disponible) {
        this.disponible = disponible;
    }

    public Grupo() {
        numero = 0;
        horario = "";
        estudiantes = new ArrayList();
        profesor = new Profesor();
        curso = new Curso();
        ciclo=new Ciclo();
        this.cupo=0;
        this.disponible=0;
        this.idEntidad=0;
    }

    public Curso getCurso() {
        return curso;
    }

    public void setCurso(Curso curso) {
        this.curso = curso;
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public int getCupo() {
        return cupo;
    }

    public void setCupo(int cupo) {
        this.cupo = cupo;
    }

    public String getHorario() {
        return horario;
    }

    public void setHorario(String horario) {
        this.horario = horario;
    }

    public List getEstudiantes() {
        return estudiantes;
    }

    public void setEstudiantes(List<Alumno> estudiantes) {
        this.estudiantes = estudiantes;
    }

    public Profesor getProfesor() {
        return profesor;
    }

    public void setProfesor(Profesor profesor) {
        this.profesor = profesor;
    }

}

File: Mobile/app/src/main/java/com/matricula/mobile/models/Grupo.kt
package com.sistema.logicaDeNegocio

import com.matricula.mobile.models.Alumno


/**
 *
 * @author DavidTK1198
 */
class Grupo {
    var numero: Int
    private var ciclo: Ciclo
    var horario: String
    private var estudiantes: List<Alumno?>
    private var profesor: Profesor
    var curso: Curso
    var idEntidad = 0
    var cupo: Int
    var disponible: Int

    constructor(
        cupo: Int,
        disponible: Int,
        numero: Int,
        horario: String,
        ciclo: Ciclo,
        profesor: Profesor,
        curso: Curso
    ) {
        this.numero = numero ////
        this.horario = horario
        this.profesor = profesor
        this.curso = curso
        this.estudiantes = ArrayList<Alumno>()
        this.ciclo = ciclo
        this.cupo = cupo
        this.disponible = cupo
    }

    fun getCiclo(): Ciclo {
        return ciclo
    }

    fun setCiclo(ciclo: Ciclo) {
        this.ciclo = ciclo
    }

    constructor() {
        numero = 0
        horario = ""
        estudiantes = ArrayList<Alumno>()
        profesor = Profesor()
        curso = Curso()
        ciclo = Ciclo()
        cupo = 0
        disponible = 0
        idEntidad = 0
    }

    fun getEstudiantes(): List<*> {
        return estudiantes
    }

    fun setEstudiantes(estudiantes: List<Alumno?>) {
        this.estudiantes = estudiantes
    }

    fun getProfesor(): Profesor {
        return profesor
    }

    fun setProfesor(profesor: Profesor) {
        this.profesor = profesor
    }
}